### PR TITLE
Add architectural decision records

### PR DESCRIPTION
**Pull request by stefaniuk**
@jope1 and @mabe13 Please, check this PR. We need to start documenting our technical decisions.

### CODE REVIEW

File: documentation/adr/005-domain_model_and_rest_interface.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+- Rest Interface (views and controllers)
+- Serialisation (the broker between the request and the domain model; and the domain model and the response)
+- Domain Model (where and how the information is kept)
+
+The Capacity Status API is required to:
+
+- Take in capacity status information to update a specified with, containing the following information:
+  - Capacity Status (the capacity status to put the service into in readable format: Red, Amber, Green)
+  - Reset Status In Minuets (when setting the capacity status to either a Red or Amber state, this is the number of minuets the service will persist in this state. After this time, the service will be automatically set back to the Green capacity status)
```
Comment (mabe13) [inline documentation/adr/005-domain_model_and_rest_interface.md]: typo 'Minuets'

[the remark this answers is not in the document]
Comment (jope1) [reply documentation/adr/005-domain_model_and_rest_interface.md]: Done

File: documentation/adr/005-domain_model_and_rest_interface.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+The Capacity Status API will be built using the Django web framework. The framework offers a number of modules and features that can be utilised when building an API. This ADR discusses these concepts and the decisions made in determining which Django features to use, and how we can use them. Essentially, this comes down to three distinct parts:
+
+- Rest Interface (views and controllers)
+- Serialisation (the broker between the request and the domain model; and the domain model and the response)
+- Domain Model (where and how the information is kept)
+
+The Capacity Status API is required to:
+
+- Take in capacity status information to update a specified with, containing the following information:
```
[the remark this answers is not in the document]
Comment (jope1) [reply documentation/adr/005-domain_model_and_rest_interface.md]: Done

File: documentation/adr/005-domain_model_and_rest_interface.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+For the interface, we decided upon using the Django Rest Framework extension (DRF). This extension provides a vast number of useful modules and components which form part of a modern RESTful API, and by leveraging these components we keep our code clean, understandable, and easy to maintain. In addition, it is very easy to plugin API documentation into this framework, the majority of the documentation being automatically generated from code, therefore keeping documentation maintenance down to a minimum.
+
+The extension comes with already defined 'views' which automatically configure (and give) the API any set of the standard REST endpoints that we would require. The Capacity Status API will require a GET and PUT endpoint, which can be automatically configured using the DRF RetrieveUpdateAPIView view. The default endpoint functionality provided by the DRF can also be easily modified to suit the specific business needs of the API.
+
+
+### Serialisation
+
+Django provides a component known as a 'Serilizer' which deals with everything concerning the transfer of data to and from the client (request/response) to and from the domain model. The serialiser deals with:
```
[the remark this answers is not in the document]
Comment (jope1) [reply documentation/adr/005-domain_model_and_rest_interface.md]: Done

File: documentation/adr/005-domain_model_and_rest_interface.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+The part of the model relating to user permissions to be able to update capacity status information also resides in the Core DoS database, and retrieval of this data will be performed by a dedicated 'DoS' module that the Capacity Status API can call. This reduces coupling concerns between Core DoS and the API, thus maintaining the integrity of the APIs core code base.
+
+Finally, the User Management model is created and maintained by the API and is stored in its own database. This further decouples the concept of the API user (or more specifically the API Key) from what is know as the user in Core DoS. The approach taken is for Django to manage the data model, and this will be performed by Django's built in database migration tools. Using this approach we are completely decoupled from what type of database we are using as Django will automatically apply the correct drivers to whatever database we choose. This also means that no database scripting of any kind is required, and the API developers need only to maintain the Django models within code. (Django will take care of applying the changes to the database).
```
[the remark this answers is not in the document]
Comment (jope1) [reply documentation/adr/005-domain_model_and_rest_interface.md]: I think this is implementation detail. Would you agree?

**Pull request merged**